app/screens/interactive_dialog/interactive_dialog.js

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {intlShape} from 'react-intl';
+import {ScrollView} from 'react-native';
+
+import {checkDialogElementForError, checkIfErrorsMatchElements} from 'mattermost-redux/utils/integration_utils';
+
+import {makeStyleSheetFromTheme} from 'app/utils/theme';
+
+import StatusBar from 'app/components/status_bar';
+import FormattedText from 'app/components/formatted_text';
+
+import DialogElement from './dialog_element.js';
+
+export default class InteractiveDialog extends PureComponent {
+    static propTypes = {
+        url: PropTypes.string.isRequired,
+        callbackId: PropTypes.string,
+        elements: PropTypes.arrayOf(PropTypes.object).isRequired,
+        notifyOnCancel: PropTypes.bool,
+        state: PropTypes.string,
+        navigator: PropTypes.object,
+        theme: PropTypes.object,
+        actions: PropTypes.shape({
+            submitInteractiveDialog: PropTypes.func.isRequired,
+        }).isRequired,
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+
+        props.navigator.setOnNavigatorEvent(this.onNavigatorEvent);
+
+        const values = {};
+        props.elements.forEach((e) => {
```

Comment:
as I mentioned above, be careful here cause you may be setting `elements` as undefined